Incorrect `group_id` matching in API RBAC

|Wazuh version|Component|
|---|---|
| Tested in 4.1.5 and master | API |

## Description

Hello team,

A user of the community has reported us the following error, which I have been able to verify. In an environment with the following groups and agents: 
|group_id|agents|
|--|--|
|group-2|001, 002|
|group-3|003|
|group-20|004|
|group-21|005|
|group-22|006|

If we create a policy like the one below:
```JSON
{
    "name": "p_group2",
    "policy": {
        "actions": [
        "agent:read"
        ],
        "resources": [
        "agent:group:group-2"
        ],
        "effect": "allow"
    }
}
```

we are able to obtain the following agents (`GET /agents?select=id,group`): 
```JSON
{
  "data": {
    "affected_items": [
      {
        "id": "001",
        "group": [
          "default",
          "group-2"
        ]
      },
      {
        "id": "002",
        "group": [
          "group-2"
        ]
      },
      {
        "id": "004",
        "group": [
          "group-20"
        ]
      },
      {
        "id": "005",
        "group": [
          "group-21"
        ]
      },
      {
        "id": "006",
        "group": [
          "group-22"
        ]
      }
    ],
    "total_affected_items": 5,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected agents information was returned",
  "error": 0
}
```

As you can see there, RBAC is not working as it should in this case. Although only agents 001 and 002 should have been returned (belonging to `group-2`), all those belonging to a group that partially matches `group-2` are obtained too. 

## Checks

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)
<!-- If changes are made to any of the following components, uncomment the corresponding line 
- [x] **Integration tests** without failures for API configuration (`/wazuh-qa/tests/integration/test_api/test_config/`)
- [x] **System tests** for agent enrollment process (`/wazuh-qa/tests/system/test_cluster/test_agent_enrollment`)
- [x] **System tests** for agent info sync process in cluster (`/wazuh-qa/tests/system/test_cluster/test_agent_info_sync`)
- [x] **System tests** for agent key polling (`/wazuh-qa/tests/system/test_cluster/test_agent_key_polling`)
- [x] **System tests** for JWT invalidation (`/wazuh-qa/tests/system/test_jwt_invalidation`)
-->

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

Best regards,
Selu.

## Status update
Hello team,

We have found the error behind this issue. We were trying to find the `group_id` within a full string with all the groups an agent belongs to. Something like this:

```python
'group-2' in 'group-21,group22\n'
```

We did not catch this error because our unit tests did not have this test case. We have updated them:

```shellsession
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/vicferpoy/Desktop/Git/wazuh/framework
plugins: cov-2.12.0, asyncio-0.15.1
collected 166 items                                                                                                                                                                         

wazuh/core/tests/test_agent.py ...................................................................................................................................................... [ 90%]
.....F..........                                                                                                                                                                      [100%]

========================================================================================= FAILURES ==========================================================================================
________________________________________________________________________ test_expand_group[group2-expected_agents1] _________________________________________________________________________

group = 'group2', expected_agents = {'001'}

    @pytest.mark.parametrize('group, expected_agents', [
        ('group1', {'000'}),
        ('group2', {'001'}),
        ('group21', {'006'}),
        ('*', {'000', '001', '002', '005', '006'})
    ])
    def test_expand_group(group, expected_agents):
        """Test that expand_group() returns expected agent IDs
    
        Parameters
        ----------
        group : str
            Name of the group to be expanded
        expected_agents : set
            Expected agent IDs for the selected group
        """
        # Clear and set get_agents_info cache
        reset_context_cache()
        test_get_agents_info()
    
        id_groups = {'000': 'group1', '001': 'group2', '002': 'group3', '004': '', '005': 'group3,group4', '006': 'group21'}
        agent_groups = os.path.join(test_data_path, 'agent-groups')
    
        with patch('wazuh.core.common.groups_path', new=agent_groups):
            try:
                os.makedirs(agent_groups)
                for id_, groups in id_groups.items():
                    with open(os.path.join(agent_groups, id_), 'w+') as f:
                        f.write(groups)
    
                result = expand_group(group)
>               assert result == expected_agents
E               AssertionError: assert {'001', '006'} == {'001'}
E                 Extra items in the left set:
E                 '006'
E                 Use -v to get the full diff

wazuh/core/tests/test_agent.py:1579: AssertionError

During handling of the above exception, another exception occurred:

group = 'group2', expected_agents = {'001'}

    @pytest.mark.parametrize('group, expected_agents', [
        ('group1', {'000'}),
        ('group2', {'001'}),
        ('group21', {'006'}),
        ('*', {'000', '001', '002', '005', '006'})
    ])
    def test_expand_group(group, expected_agents):
        """Test that expand_group() returns expected agent IDs
    
        Parameters
        ----------
        group : str
            Name of the group to be expanded
        expected_agents : set
            Expected agent IDs for the selected group
        """
        # Clear and set get_agents_info cache
        reset_context_cache()
        test_get_agents_info()
    
        id_groups = {'000': 'group1', '001': 'group2', '002': 'group3', '004': '', '005': 'group3,group4', '006': 'group21'}
        agent_groups = os.path.join(test_data_path, 'agent-groups')
    
        with patch('wazuh.core.common.groups_path', new=agent_groups):
            try:
                os.makedirs(agent_groups)
                for id_, groups in id_groups.items():
                    with open(os.path.join(agent_groups, id_), 'w+') as f:
                        f.write(groups)
    
                result = expand_group(group)
                assert result == expected_agents
            except Exception as e:
>               pytest.fail(f'Exception raised: {e}')
E               Failed: Exception raised: assert {'001', '006'} == {'001'}
E                 Extra items in the left set:
E                 '006'
E                 Use -v to get the full diff

wazuh/core/tests/test_agent.py:1581: Failed
================================================================================== short test summary info ==================================================================================
FAILED wazuh/core/tests/test_agent.py::test_expand_group[group2-expected_agents1] - Failed: Exception raised: assert {'001', '006'} == {'001'}
========================================================================= 1 failed, 165 passed, 9 warnings in 1.22s =========================================================================

```

Using a very similar environment as the one described in the issue (we only have read permissions for group `group-2`), we had this response before:

```json
{
  "data": {
    "affected_items": [
      {
        "group": [
          "default",
          "group-2"
        ],
        "id": "001"
      },
      {
        "group": [
          "default",
          "group-2"
        ],
        "id": "002"
      },
      {
        "group": [
          "default",
          "group-20"
        ],
        "id": "003"
      },
      {
        "group": [
          "default",
          "group-21"
        ],
        "id": "004"
      }
    ],
    "total_affected_items": 4,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected agents information was returned",
  "error": 0
}
```

After the fix, this is what we get:

```json
{
  "data": {
    "affected_items": [
      {
        "id": "001",
        "group": [
          "default",
          "group-2"
        ]
      },
      {
        "id": "002",
        "group": [
          "default",
          "group-2"
        ]
      }
    ],
    "total_affected_items": 2,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected agents information was returned",
  "error": 0
}
```

Regards,
Víctor

--- a/framework/wazuh/core/agent.py
+++ b/framework/wazuh/core/agent.py
@@ -1250,7 +1250,7 @@ def expand_group(group_name):
             try:
                 with open(path.join(common.groups_path, file), 'r') as f:
                     file_content = f.readlines()
-                len(file_content) == 1 and group_name in file_content[0] and agents_ids.add(file)
+                len(file_content) == 1 and group_name in file_content[0].split(',') and agents_ids.add(file)
             except FileNotFoundError:
                 # Agent group removed while running through listed dir
                 pass

--- a/framework/wazuh/core/tests/test_agent.py
+++ b/framework/wazuh/core/tests/test_agent.py
@@ -1548,7 +1548,8 @@ def test_get_groups():
 @pytest.mark.parametrize('group, expected_agents', [
     ('group1', {'000'}),
     ('group2', {'001'}),
-    ('*', {'000', '001', '002', '005'})
+    ('group21', {'006'}),
+    ('*', {'000', '001', '002', '005', '006'})
 ])
 def test_expand_group(group, expected_agents):
     """Test that expand_group() returns expected agent IDs
@@ -1564,7 +1565,7 @@ def test_expand_group(group, expected_agents):
     reset_context_cache()
     test_get_agents_info()
 
-    id_groups = {'000': 'group1', '001': 'group2', '002': 'group3', '004': '', '005': 'group3,group4', '006': ''}
+    id_groups = {'000': 'group1', '001': 'group2', '002': 'group3', '004': '', '005': 'group3,group4', '006': 'group21'}
     agent_groups = os.path.join(test_data_path, 'agent-groups')
 
     with patch('wazuh.core.common.groups_path', new=agent_groups):

